### DISCUSSION TITLE
Stuck on intel

### DISCUSSION THREAD

**Discussion by KaJashey** [Homebrew/discussions#5588] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 2 days ago
Core tap JSON: 04 Sep 15:58 UTC
Core cask tap JSON: 04 Sep 15:58 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: true
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  neofetch
  openssl@1.1

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@3.12

Warning: Your Homebrew's prefix is not /opt/homebrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

I have an M1 mac. my terminal is arm. the results of the arch command are "arm64" yet when i install homebrew I get intel homebrew. Not sure what's going on.

I've uninstalled and reinstalled to to avail

**Comment by KaJashey** [comment]
Fixed it myself. I had to go to /usr/local and delete homebrew folder and the cellar folder for good measure. Shame that the uninstall script doesn't get those.

Now I have arm64 in brew config and and ready to go in brew doctor.

**Reply by cho-m** [reply, marked as answer]
The uninstall script operates based on the current `arch`, which can be overridden with `--path` as recommended at https://github.com/Homebrew/install/blob/master/README.md#uninstall-homebrew
> If you want to to uninstall Homebrew from a specific prefix (e.g. when migrating from Intel to Apple Silicon processors), download the uninstall script and run it with `--path`

---

Also see https://docs.brew.sh/Common-Issues#unintentional-dual-homebrew-installations for recommendation on migration from Intel to ARM.

**Reply by KaJashey** [reply]
Thank you. Is there any way to update the script so it uninstalls both locations as default? That behavior would be more in line of what I would expect from an uninstaller.

**Reply by gromgit** [reply]
> Is there any way to update the script so it uninstalls both locations as default? That behavior would be more in line of what I would expect from an uninstaller.

It would also greatly upset folks who install ARM64 Homebrew alongside Intel Homebrew to do a proper migration, and then run the uninstall script. I think "WTF!!!" would be the least vitriolic reaction when _both_ installations disappear after all that hard work.

**Comment by p-linnane** [comment]
At the bottom of your `brew config` you'll see
```
Rosetta 2: true
```
Meaning your terminal is running in x86_64 mode, which is why everything is pointing to the Intel default prefix.
